src/App.tsx

```diff
@@ -76,8 +199,8 @@ export const App: React.FC = () => {
                 {author && loaded && !hasError && posts.length > 0 && (
                   <PostsList
                     posts={posts}
-                    selectedPostId={selectedPost?.id}
-                    onPostSelected={setSelectedPost}
+                    selectedPostId={currentPost?.id}
```

Comment:
Ensure that `selectedPostId` is correctly derived from `currentPost?.id`. If `currentPost` is null, this will pass `undefined`, which is likely intended, but double-check that this behavior is handled correctly in `PostsList`.